app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -49,23 +44,35 @@ def xms_mirid_hash
       # according to fields we need to retrieve from the claim.
       # ultimately, transforming "/key1/value1/key2/value2" to {"key1" => "value1", "key2" => "value2"}
       def parsed_xms_mirid
-        split_xms_mirid = @xms_mirid_token_field.split('/')
+        begin
+          split_xms_mirid = @xms_mirid_token_field.split('/')
 
-        if split_xms_mirid.first == ''
-          split_xms_mirid = split_xms_mirid.drop(1)
-        end
+          if split_xms_mirid.first == ''
+            split_xms_mirid = split_xms_mirid.drop(1)
+          end
 
-        index = 0
-        split_xms_mirid.each_with_object({}) do |property, xms_mirid_hash|
-          case property
-          when "subscriptions"
-            xms_mirid_hash["subscriptions"] = split_xms_mirid[index + 1]
-          when "resourcegroups"
-            xms_mirid_hash["resourcegroups"] = split_xms_mirid[index + 1]
-          when "providers"
-            xms_mirid_hash["providers"] = split_xms_mirid[index + 1, index + 3]
+          index = 0
+          split_xms_mirid.each_with_object({}) do |property, xms_mirid_hash|
+            case property
+            when "subscriptions"
+              xms_mirid_hash["subscriptions"] = split_xms_mirid[index + 1]
+            when "resourcegroups"
+              xms_mirid_hash["resourcegroups"] = split_xms_mirid[index + 1]
+            when "providers"
+              xms_mirid_hash["providers"] = split_xms_mirid[index + 1, index + 3]
+            end
+            index += 1
           end
-          index += 1
+        rescue => e
+          raise Err::XmsMiridParseError.new(@xms_mirid_token_field, e.inspect)
+        end
+      end
+
+      def validate_xms_mirid_format
+        required_keys = %w(subscriptions resourcegroups providers)
+        missing_keys = required_keys - xms_mirid_hash.keys
+        unless missing_keys.empty?
+          raise Err::MissingRequiredKeysInXmsMirid.new(missing_keys, @xms_mirid_token_field)
```

Comment:
`MissingRequiredKeysInXmsMirid` please add this to the log issue for TWs
https://github.com/cyberark/conjur-docs/issues/742
Create a new section "Added logs" to the end so Shuli knows which new ones to review